### PR TITLE
Add compatibility layer for Linux kernel versions

### PR DESCRIPTION
**Pull request by AntoineBlaud**
### Overview

This pull request introduces a compatibility layer to address structural changes in the Linux kernel starting from version 5.17. Notably, modifications have been made to the `page` and `slab` structures, necessitating adjustments in codebase.

### CODE REVIEW

**Comment by fidgetingbits** [comment]
Thanks! I'll try to find time to review and test it tomorrow.

File: libslub/frontend/commands/gdb/sbtrace.py [hunk 1]
```diff
@@ -56,6 +58,10 @@ def invoke(self, arg, from_tty):
         """
 
         log.debug("sbtrace.invoke()")
+        
+        if len(self.args.names) == 0:
```
Comment (fidgetingbits) [inline libslub/frontend/commands/gdb/sbtrace.py]: I guess this is related to a separate fix. I don't mind to keep it but you can feel free to send a separate PR if you want.

File: libslub/pyslub.py [hunk 2]
```diff
@@ -35,8 +37,12 @@ def __init__(self):
             breakpoints_enabled = config.getboolean("Slab", "breakpoints_enabled")
         except configparser.NoOptionError:
             breakpoints_enabled = False
+            
+        # read the current kernel version using gdb
+        version_string = gdb.parse_and_eval("(char *)linux_banner").string()
+        release = re.search(r'(\d+\.\d+\.\d+)', version_string).group(1)
```
Comment (fidgetingbits) [inline libslub/pyslub.py]: Given that this is duplicated code from earlier probably just makes sense to make it like a helper function or something and call it in both places. I don't mind to do it later if you don't want to do it now though.

**Review by fidgetingbits — APPROVED** [review]
This seems to work great. I've tested it on something earlier and later then 5.17. I'll wait a little bit to see if you respond about wanting to use a single helper function, but if you don't want to change that I will merge it as is and address it later.

File: libslub/pyslub.py (hunk 2 above)
Comment (AntoineBlaud) [reply libslub/pyslub.py]: Hello @fidgetingbits,

Following your guidance, I've created a second commit to address the duplicate code issue.

**Pull request merged**